packages/eas-cli/src/update/queries.ts

```diff
@@ -0,0 +1,104 @@
+import chalk from 'chalk';
+import CliTable from 'cli-table3';
+
+import { PaginatedQueryFlags } from '../commandUtils/pagination';
+import { ViewBranchUpdatesQuery } from '../graphql/generated';
+import { UpdateQuery } from '../graphql/queries/UpdateQuery';
+import Log from '../log';
+import formatFields from '../utils/formatFields';
+import { printJsonOnlyOutput } from '../utils/json';
+import { PaginatedQueryResponse, paginatedQueryWithConfirmPromptAsync } from '../utils/queries';
+import { UPDATE_COLUMNS, formatUpdate, getUpdateGroupsWithPlatforms } from './utils';
+
+type UpdateBranch = Exclude<
+  ViewBranchUpdatesQuery['app']['byId']['updateBranchByName'],
+  undefined | null
+>;
+
+export const UPDATES_LIMIT = 50;
+
+export async function listAndRenderUpdatesOnBranchByNameAsync(
+  projectId: string,
+  branchName: string,
+  flags: PaginatedQueryFlags
+): Promise<void> {
+  if (flags.nonInteractive || flags.limit) {
+    const { queryResponse: updates } = await queryUpdateGroupsForBranchAsync(
+      flags.limit ?? UPDATES_LIMIT,
+      flags.offset ?? 0,
+      projectId,
+      branchName
+    );
+    renderUpdateGroups(updates, flags);
+  } else {
+    await paginatedQueryWithConfirmPromptAsync({
+      pageSize: flags.limit ?? UPDATES_LIMIT,
+      offset: flags.offset ?? 0,
+      queryToPerform: (pageSize: number, offset: number) =>
```

Comment:
I'm not 100% sure but types here should be optional. This should be also valid TS:
```suggestion
      queryToPerform: (pageSize, offset) =>
```